calico_containers/adapter/datastore.py

```diff
@@ -590,6 +596,87 @@ def remove_ip_pool(self, version, pool):
             raise KeyError("%s is not a configured IP pool." % pool)
 
     @handle_errors
+    def get_hostnames(self):
+        """
+        Return a list of hostnames
+        :return:  A list of hostnames configured in etcd.
+        """
+        # Get a list of the hosts.
+        hosts = []
+        results = self.etcd_client.read(HOSTS_PATH)
+        for result in results.children:
+            packed = result.key.split("/")
+            # Handle no hosts being configured (etcd returns the parent).
+            if len(packed) != 5:
```

Comment:
That feels more fragile to me since theoretically you could have HOSTS_PATH with and without the trailing / and get the same results from etcd ... but obviously HOSTS_PATH.split("/") would be different.

I'd quite like to sort out all of the key parsing to use actual regexes with names groups.